tests/ui_tests/reporting/html.py

```diff
@@ -92,13 +92,51 @@ def image_link(
     )
 
 
+def diff_row(
+    left: str | None,
+    right: str | None,
+    cur_dir: Path,
+    bgcolor: str,
+    hidden: bool = False,
+):
+    with tr(bgcolor=bgcolor, _class="hidden" if hidden else ""):
+        image_column(left, cur_dir)
+        image_column(right, cur_dir)
+        diff_column()
+
+
+def collapsible_rows(rows, cur_dir):
+    collapse = len(rows) > 0
+    color = "yellow" if collapse else "white"
+    for left, right in rows:
+        diff_row(left, right, cur_dir, color, collapse)
+    if collapse:
+        with tr(bgcolor="yellow"):
+            with td(colspan=3, _class="showLink"):
+                span(f"{len(rows)} hidden")
+                a("show all", href="#", onclick="showAllHidden();")
+
+
 def diff_table(diff: Iterable[tuple[str | None, str | None]], cur_dir: Path) -> None:
-    for left, right in diff:
-        if left == right:
-            background = "white"
-        else:
-            background = "red"
-        with tr(bgcolor=background, onclick="createDiff(this)"):
-            image_column(left, cur_dir)
-            image_column(right, cur_dir)
-            diff_column()
+    has_diff = False
+    rows = []
+    for l, r in diff:
+        if l != r:
+            has_diff = True
+        rows.append((l, r))
```

Comment:
perhaps more pythonic way
```suggestion
    rows = list(diff) 
    has_diff = any(l_hash != r_hash for l_hash, r_hash in rows)
```